core/sequencer/server_test.go

```diff
@@ -88,16 +97,30 @@ func (m *fakeMap) GetAndVerifyLatestMapRoot(_ context.Context) (*tpb.SignedMapRo
 
 type fakeBatcher struct {
 	highestRev int64
+	batches    map[int64]*spb.MapMetadata
 }
 
 func (b *fakeBatcher) HighestRev(_ context.Context, _ string) (int64, error) {
 	return b.highestRev, nil
 }
-func (b *fakeBatcher) WriteBatchSources(_ context.Context, _ string, _ int64, _ *spb.MapMetadata) error {
+func (b *fakeBatcher) WriteBatchSources(_ context.Context, _ string, rev int64, meta *spb.MapMetadata) error {
+	b.batches[rev] = meta
 	return nil
 }
-func (b *fakeBatcher) ReadBatch(_ context.Context, _ string, _ int64) (*spb.MapMetadata, error) {
-	return &spb.MapMetadata{}, nil
+func (b *fakeBatcher) ReadBatch(_ context.Context, _ string, rev int64) (*spb.MapMetadata, error) {
+	meta, ok := b.batches[rev]
+	if !ok {
+		return nil, fmt.Errorf("batch %v not found", rev)
+	}
+	return meta, nil
+}
+
+type fakeMapConn struct {
+	tpb.TrillianMapClient
+}
+
+func (m *fakeMapConn) GetLeavesByRevision(_ context.Context, _ *tpb.GetMapLeavesByRevisionRequest, _ ...grpc.CallOption) (*tpb.GetMapLeavesResponse, error) {
+	return nil, status.Errorf(codes.Unimplemented, "unimplemented")
```

Comment:
As per discussion, would it be possible to test that this call doesn't get duplicates indices? This would be a better indication of what this PR aims for.

Reply:
Yes, but that's the code being fixed... so unfortunately it won't be in test vector format.

Reply:
Not sure I understand. Line 316 checks that `Unimplemented` is returned. I propose to still return `Unimplemented`, but record the request in `fakeMapConn`, and then check that there are no duplicates in this request (gmock-like). As far as I understand, checking this would distinguish the old behavior from the new one.